actors/miner/src/lib.rs

```diff
@@ -3701,33 +3747,33 @@ where
                 update.replica_proof.len(),
                 update.sector_number,
             );
-            continue;
+            fail_validation = true;
         }
 
-        if update.deals.is_empty() {
+        if require_deals && update.deals.is_empty() {
             info!("must have deals to update, skipping sector {}", update.sector_number,);
```

Comment:
Should we now be requiring non zero data commitment for the new unsealed_cid?  The old deal requirement achieved this.  I don't know if it would be possible to prove update from zero data to zero data but it makes me nervous to be opening up new corner cases to the proofs API and possibly actors code.

Reply:
Well actually come to think of it there is nothing currently stopping a zero valued deal so this is actually not a new case.

Reply:
I did think about this and decided to leave it possible to prove an update zero to zero. In the future resnap will probably want to be able to prove non-zero to zero.